nav_msgs/msg/Particle.msg

```diff
@@ -0,0 +1,5 @@
+# This represents an individual particle with weight produced by a particle filter
```

Comment:
> Same if you want it to be a particle that can be used with any robot navigation particle filter

I'm not bringing this up to try to argue your point, see below I think we have some philosophical overlap on naming, but I'd like to temper expectations on what this package is. `nav_msgs`, to the best of my understanding, has never been a collection of messages suitable for every form of robot navigation. If that were the case, about 2/3 of these messages shouldn't be here. These messages are here to support the work happening in the navigation stack, which often can lend themselves useful to other navigation projects. 

Anyhow.

I think your name suggestions are really good - that brings up another side of your argument that I can agree with. This should be more specific as to the type of particle it is. Personally I think the `weighted` bit is redundant. In the general formulation of a particle filter, each particle has a score so I think that's totally implied. `PoseParticle` and `PoseParticleCloud` I could totally support. 

Other examples in the community I found (still MCLs admittedly) which could readily use this definition of the messages proposed, under whatever name we decide on:
- NDT-MCL uses a PoseParticle  
- Quick-MCL uses just a visualization_msgs/MarkerArray - no weights
- mcl pi gazebo uses the PoseArray that we're trying to move from to show particle weights
- MRPT localization uses the PoseArray that we're trying to move from to show particle weights
- AMCL3D fails to publish anything at all

You're not being harsh, we're having a conversation - a good conversation :-)

Reply:
Ok, with the presumption that `Particle` means "state plus a single weight", then `PoseParticle*` sounds good to me.

Reply:
@naiveHobo are you OK with that?

Reply:
> nav_msgs, to the best of my understanding, has never been a collection of messages suitable for every form of robot navigation. If that were the case, about 2/3 of these messages shouldn't be here.

I agree with you, but moving forward, if I'm pressed to say "what should be in common interfaces or not" and "what should be in rviz default plugins or not", then I think it should either be "the message is generally useful for running a ROS system" or "the message is general enough to be used by multiple parties for the same purpose". Many of the messages here are here due to historical reasons.

We probably need a "generic nav messages package" and a "ROS nav(2) project specific messages package" to separate the use cases. Like many things in ROS the nav stack's messages were intended to be the standard, but it fell short.

Reply:
`PoseParticle` works. I'll go ahead with the changes then?

Reply:
> We probably need a "generic nav messages package" and a "ROS nav(2) project specific messages package" to separate the use cases. Like many things in ROS the nav stack's messages were intended to be the standard, but it fell short.

We sort of have that, we have nav2_msgs that are messages in a staging ground before eventually being homologated with `nav_msgs`. Whether that's `nav2_msgs` go into `nav_msgs` or `nav_msgs` gets relocated to `ros-perception`. The other things in nav2_msgs are still changing and don't belong here (yet). I suspect the actions will never belong here unless we move to `ros-perception` out of `common_interfaces` to signal the more specific project nature of it. I'm not pushing for that right now or in the near term future.

In any case, hopefully over the next few years of navigation work, we'll start being able to generalize things better. I'm slowly breaking up alot of the early assumptions and making things more flexible, but it takes time. 

@naiveHobo, yes please make those updates.

Reply:
> These messages are here to support the work happening in the navigation stack

I've never thought about the `nav_msgs` package this way before. Though indeed it is written on the [wiki](http://wiki.ros.org/nav_msgs) and in the README of this package that they are for *the* navigation stacks. Personally, I've used them in several other projects that have not used any of the other elements of the nav stack (ROS 1). It might be worth considering removing the emphasis on any particular navigation stack.

+1 for renaming to `PoseParticle` and `PoseParticleCloud`.

I think it would go a long way to get buy-in from maintainers of the other packages that produce similar data.
We can also elicit general feedback from the community with a Discourse post.